pkg/sdn/plugin/egress_network_policy.go

```diff
@@ -108,8 +108,11 @@ func (plugin *OsdnNode) syncEgressDNSPolicyRules() {
 			continue
 		}
 
-		plugin.egressPoliciesLock.Lock()
-		defer plugin.egressPoliciesLock.Unlock()
-		plugin.updateEgressNetworkPolicyRules(vnid)
+		func() {
```

Comment:
at this point, why not just not use defer?  Just lock, <op>, unlock would be simpler IMHO.